### ISSUE TITLE
Display virtualtext on each error/warning, not only current

### ISSUE THREAD

**Issue by GoldsteinE** [neomake/neomake#2446]
Subj. Internal NeoVim's LSP client shows virtualtext descriptions on each warning, so it should be possible.

**Comment by blueyed** [comment]
You can try to call `:call neomake#virtualtext#show_errors()` manually.
I've thought for it to be less distracting to only display the current error.

**Comment by GoldsteinE** [comment]
`:call neomake#virtualtext#show()` works nice, thanks!
It would be cool if it was avialable as built-in option. Maybe I'll send a PR.

**Comment by blueyed** [comment]
Cool.  It could be done by changing `neomake#virtualtext#displayhandle_current_error` to display all errors then maybe.  But it seems not trivial to refactor / change this.
And I'd rather use popup / floating windows instead then, which allows to display more information and for positioning them, deprecating/removing virtualtext support then.

**Comment by GoldsteinE** [comment]
I added this to my vimrc:
```vim
autocmd User NeomakeJobFinished silent call neomake#virtualtext#hide() | call neomake#virtualtext#show()
```
but it doesn't show info messages.
One big usecase for virtualtext on all diagnostics is messages referring other messages, like this:
![image](https://user-images.githubusercontent.com/12019211/71356724-1cfdbd00-2594-11ea-8438-51d3eae82a42.png)

**Comment by GoldsteinE** [comment]
UPD: it's probably not because of message type, but because clangcheck entry doesn't contain bufnr.

**Comment by blueyed** [comment]
> UPD: it's probably not because of message type, but because clangcheck entry doesn't contain bufnr.

Yeah.  Check the output of `:echo getloclist(0)` to see where the entries differ - if there's an entry for the "previous definition is here" message after all.

I think you might be better off using LSP for this in general (only) for now.

**Comment by GoldsteinE** [comment]
Seems like only first message is shown
![image](https://user-images.githubusercontent.com/12019211/71359729-9d292000-259e-11ea-9536-7b3bb6bf6513.png)

**Comment by FelipeLema** [comment]
you may want to check out https://codeberg.org/FelipeLema/neomake-diagnostics.nvim if provided solutions do not seem enough

xref https://github.com/neomake/neomake/issues/2554#issuecomment-1861252541
